README.md
# lorem_picsum

[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT) 
[![pub package](https://img.shields.io/pub/v/lorem_picsum.svg)](https://pub.dartlang.org/packages/lorem_picsum)


Dart client for Lorem Picsum, the Lorem Ipsum for photos

## Usage

```dart
import 'package:lorem_picsum/lorem_picsum.dart';

main() async {
  final loremPicsum = LoremPicsum();

  // get random image url, provide the width and height
  String image = loremPicsum.getImage(200, 300);

  // get specific image, provide id, width, and height
  String specificImage = loremPicsum.getSpecificImage(1, 200, 300);

  // get the same random image every time, provide seed, width, and height
  String staticImage = loremPicsum.getStaticImage('seed', 200, 300);

  // get image list, default limit is 30
  List<Photo> photoList = await loremPicsum.getImageList(limit: 30, page: 1);

  // get image detail
  Photo photo = await loremPicsum.getImageDetail(1);
}
```

## License

MIT

## Credits

This package is based on [Lorem Picsum](https://picsum.photos/) created by [David Marby](https://dmarby.se/) and [Nijiko Yonskai](https://github.com/Nijikokun)

CHANGELOG.md
## 0.0.1+1

* Remove unused dependencies.

## 0.0.1

* Initial release.


test/lorem_picsum_test.dart
import 'package:test/test.dart';
import 'package:lorem_picsum/lorem_picsum.dart';

void main() {
  LoremPicsum loremPicsum;

  setUpAll(() {
    loremPicsum = LoremPicsum();
  });
  test('get image list', () async {
    var imageList = await loremPicsum.getImageList();
    expect(imageList.length, 30);
    expect(imageList, isA<List<Photo>>());
  });

  test('get image detail', () async {
    var image = await loremPicsum.getImageDetail(0);
    expect(image.id, 0);
    expect(image, isA<Photo>());
  });
}


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter_test/flutter_test.dart';

void main() {
  testWidgets('-', (WidgetTester tester) async {
  });
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:lorem_picsum/lorem_picsum.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Example',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  final LoremPicsum _loremPicsum = LoremPicsum();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Lorem Picsum Example')),
      body: SingleChildScrollView(
        child: Column(
          children: [
            Wrap(
              children: [
                Image.network(_loremPicsum.getImage(200, 200)),
                Image.network(_loremPicsum.getImage(300, 200, grayscale: true)),
                Image.network(_loremPicsum.getImage(200, 400, blur: 5)),
                Image.network(
                    _loremPicsum.getSpecificImage(1, 200, 200, blur: 1)),
              ],
            ),
          ],
        ),
      ),
    );
  }
}


lib/src/lorem_picsum.dart
import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:lorem_picsum/src/models/photo.dart';

class LoremPicsum {
  final String _baseUrl = 'picsum.photos';

  Future<dynamic> _getRequest(String url, {Map<String, String> query}) async {
    var uri = Uri.https(_baseUrl, url, query);
    var response = await http.get('$uri');
    if (response.statusCode == 200) {
      print('response : ${response.body}');
      return response.body;
    } else {
      throw Exception('Request failed with status: ${response.statusCode}.');
    }
  }

  /// Get a list of photos.
  ///
  /// The API will return 30 items per page by default.
  /// To request another page, use the [page] parameter.
  /// To change the amount of items per page, use the [limit] parameter.
  Future<List<Photo>> getImageList({int page = 1, int limit = 30}) async {
    var url = '/v2/list';
    var query = {'page': '$page', 'limit': '$limit'};
    var response = await _getRequest(url, query: query);

    var photoList = <Photo>[];
    List json = jsonDecode(response);
    json.forEach((e) {
      var photo = Photo.fromJson(e);
      photoList.add(photo);
    });

    return photoList;
  }

  /// Get detailed information about a specific image by providing [id].
  Future<Photo> getImageDetail(int id) async {
    var url = '/id/$id/info';
    var response = await _getRequest(url);
    return Photo.fromJson(jsonDecode(response));
  }

  /// get random image.
  ///
  /// to apply grayscale effect, set [grayscale] to `true`.
  /// to apply blur effect, provide an `integer` between `1` and `10`.
  String getImage(int width, int height, {bool grayscale = false, int blur}) {
    var url = '/$width/$height';
    var query = {
      if (grayscale) 'grayscale': '',
      if (blur != null) 'blur': '$blur'
    };
    var uri = Uri.https(_baseUrl, url, query);
    return uri.toString();
  }

  /// get specific image by providing [id].
  ///
  /// to apply grayscale effect, set [grayscale] to `true`.
  /// to apply blur effect, provide an `integer` between `1` and `10`.
  String getSpecificImage(int id, int width, int height,
      {bool grayscale = false, int blur}) {
    var url = '/id/$id/$width/$height';
    var query = {
      if (grayscale) 'grayscale': '',
      if (blur != null) 'blur': '$blur'
    };
    var uri = Uri.https(_baseUrl, url, query);
    return uri.toString();
  }

  /// Get the same random image every time based on a seed, by providing [seed].
  ///
  /// to apply grayscale effect, set [grayscale] to `true`.
  /// to apply blur effect, provide an `integer` between `1` and `10`.
  String getStaticImage(String seed, int width, int height,
      {bool grayscale = false, int blur}) {
    var url = '/seed/$seed/$width/$height';
    var query = {
      if (grayscale) 'grayscale': '',
      if (blur != null) 'blur': '$blur'
    };
    var uri = Uri.https(_baseUrl, url, query);
    return uri.toString();
  }
}
